src/app/campaign-list/list/filter-facet.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, Input, Output, EventEmitter, ChangeDetectionStrategy } from '@angular/core';
+import { Facet } from '../campaign-list.service';
+
+@Component({
+  selector: 'grove-filter-facet',
+  template: `
+    <mat-form-field>
+      <mat-select
+        formControlName="status"
+        placeholder="Filter by {{facetName}}"
+        [multiple]="multiple"
+        [(value)]="selectedOptions">
+        <mat-option *ngIf="!multiple">None</mat-option>
+        <mat-option *ngFor="let option of options" [value]="option.id">{{ option.label }}</mat-option>
+      </mat-select>
+    </mat-form-field>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class FilterFacetComponent {
+  @Input() facetName: string;
+  @Input() multiple: boolean;
+  @Input() options: Facet[];
+  @Input()
+  set selectedOptions(values: number | string | number[] | string[]) {
+    this._selectedOptions = values;
+    this.selectedOptionsChange.emit(values);
```

Comment:
This is a cool way to emit whenever the field changes... may steal this...

Reply:
yeah, seemed like I had to use this here because the material select uses that double binding on the value